test/conftest.py

```diff
@@ -167,3 +171,11 @@ def pytest_terminal_summary(terminalreporter, exitstatus, config):
                     terminalreporter._outrep_summary(rep)
                     terminalreporter._handle_teardown_sections(rep.nodeid)
     yield
+
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_pycollect_makemodule(module_path, path, parent) -> Module:
```

Comment:
You should write a note on how pytest hook works and how to write one to share with us :D

I guess what this mean is to make pytest returns a test method belonging to `__main__` instead of the file name like `test_ops_fwd_gradients.py`

Reply:
pytest has a lot of documentation about writing hooks at https://docs.pytest.org/en/7.1.x/how-to/writing_hook_functions.html.  I also just look for examples in the pytest plugins whenever I need to write a hook 

yeah thats pretty much it

Reply:
Bookmarked :)